### PR TITLE
Build Script Refactor

### PR DESCRIPTION
**Pull request by tomspilman**
This PR refactors the main build script to allow tests, docs, and installers to run without target dependencies.  This will allow us to do some restructuring of the build server into more parallel steps.

### CODE REVIEW

**Comment by tomspilman** [comment]
@dellis1972 - Give this a review and merge when you can.

**Comment by tomspilman** [comment]
Note that old PRs need to merge this commit and/or rebase/merge to make the new build process work.

**Comment by tomspilman** [comment]
Once merged I need to fix up the website links to point to the new locations of the develop installers.

File: default.build [hunk 1]
```diff
@@ -181,35 +194,38 @@
 
         <exec program="${msbuild14dir}\msbuild.exe " commandline='MonoGame.Framework.WindowsUniversal.sln /t:Clean /p:Configuration=Release /p:Platform="Any CPU"' />
         <exec program="${msbuild14dir}\msbuild.exe " commandline='MonoGame.Framework.WindowsUniversal.sln /t:Build /p:Configuration=Release /p:Platform="Any CPU"' />
-        <exec program="C:\Windows\system32\cmd.exe" commandline='/C rmdir /S /Q WindowsUAP' workingdir='MonoGame.Framework\bin' failonerror='false' />
-        <exec program="C:\Windows\system32\cmd.exe" commandline='/C rename WindowsUniversal WindowsUAP' workingdir='MonoGame.Framework\bin' />
```
Comment (dellis1972) [inline default.build]: @tomspilman do we not need this rename anymore?

File: Installers/default.build [hunk 2]
```diff
@@ -19,35 +19,28 @@
...
       <if test="${file::exists('C:\Program Files (x86)\NSIS\makensis.exe')}">
-
         <echo append="false" file="Windows/header.nsh">
 !define FrameworkPath "${project::get-base-directory()}"
 !define VERSION "3.0"
 !define INSTALLERVERSION "${buildNumber}"
         </echo>
-        <call target="download"/>
-        <exec program="C:\Windows\system32\cmd.exe" commandline='/C rmdir /S /Q WindowsUniversal' workingdir="${project::get-base-directory()}\..\MonoGame.Framework\bin" failonerror='false' />
-		<exec program="C:\Windows\system32\cmd.exe" commandline='/C rename WindowsUAP WindowsUniversal' workingdir="${project::get-base-directory()}\..\MonoGame.Framework\bin" />
```
Comment (dellis1972) [inline Installers/default.build]: @tomspilman same question. Do we not need this?

**Comment by dellis1972** [comment]
@tomspilman  looks fine apart from those two Windows Universal removals.

File: default.build (hunk 1 above)
Comment (tomspilman) [inline default.build]: Nope...  with the change in the build server I fixed the artifact copy which used to use the old name.

File: Installers/default.build (hunk 2 above)
Comment (tomspilman) [inline Installers/default.build]: Nope.

**Comment by tomspilman** [comment]
@dellis1972 - Neither are a problem... the build server settings that required the old name have been fixed.

**Comment by dellis1972** [comment]
@tomspilman ok cool in that case merging :)

**Pull request merged**